File: app/src/main/java/com/android/example/popularmovies/ui/detail/fragments/videos/VideosAdapter.java
package com.android.example.popularmovies.ui.detail.fragments.videos;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

import com.android.example.popularmovies.R;
import com.android.example.popularmovies.databinding.VideoListItemBinding;
import com.android.example.popularmovies.utils.json.MovieVideo;

import java.util.List;

public class VideosAdapter extends RecyclerView.Adapter<VideosAdapter.VideosViewHolder> {

    @Nullable
    private List<MovieVideo> mVideos;

    public interface VideoOnClickListener{
        void onListItemClick(MovieVideo video);
    }

    private VideoOnClickListener onClickListener;

    public VideosAdapter(VideoOnClickListener onClickListener) {
        this.onClickListener = onClickListener;
    }

    @NonNull
    @Override
    public VideosViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.video_list_item, parent, false);
        return new VideosViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull VideosViewHolder holder, int position) {
        if (mVideos != null) {
            holder.bind(mVideos.get(position));
        }
    }

    @Override
    public int getItemCount() {
        if (mVideos != null) {
            return mVideos.size();
        } else {
            return 0;
        }
    }

    public void setVideos(List<MovieVideo> videos){
        mVideos = videos;
        notifyDataSetChanged();
    }

    class VideosViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        VideoListItemBinding binding;

        public VideosViewHolder(@NonNull View itemView) {
            super(itemView);
            binding = VideoListItemBinding.bind(itemView);
            itemView.setOnClickListener(this);
        }

        void bind(MovieVideo video){
            binding.tvVideoTitle.setText(video.getName());
        }

        @Override
        public void onClick(View v) {
            if (mVideos != null){
                onClickListener.onListItemClick(mVideos.get(getAdapterPosition()));
            }
        }
    }
}

File: app/src/main/java/com/android/example/popularmovies/ui/detail/fragments/videos/VideosAdapter.kt
package com.android.example.popularmovies.ui.detail.fragments.videos

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.android.example.popularmovies.R
import com.android.example.popularmovies.databinding.VideoListItemBinding
import com.android.example.popularmovies.ui.detail.fragments.videos.VideosAdapter.VideosViewHolder
import com.android.example.popularmovies.utils.json.MovieVideo

class VideosAdapter(private val onClickListener: VideoOnClickListener) : RecyclerView.Adapter<VideosViewHolder>() {
    private var mVideos: List<MovieVideo>? = null

    interface VideoOnClickListener {
        fun onListItemClick(video: MovieVideo)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VideosViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.video_list_item, parent, false)
        return VideosViewHolder(view)
    }

    override fun onBindViewHolder(holder: VideosViewHolder, position: Int) {
        if (mVideos != null) {
            holder.bind(mVideos!![position])
        }
    }

    override fun getItemCount(): Int = mVideos?.size ?: 0

    fun setVideos(videos: List<MovieVideo>?) {
        mVideos = videos
        notifyDataSetChanged()
    }

    inner class VideosViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        private var binding: VideoListItemBinding = VideoListItemBinding.bind(itemView)
        fun bind(video: MovieVideo) {
            binding.tvVideoTitle.text = video.name
        }

        override fun onClick(v: View) {
            if (mVideos != null) {
                onClickListener.onListItemClick(mVideos!![adapterPosition])
            }
        }

        init {
            itemView.setOnClickListener(this)
        }
    }

}